src/lib.rs

```diff
@@ -136,22 +134,21 @@
 
 pub mod converter;
 
-mod builder;
+mod backend;
 mod character;
 mod fm_index;
-mod iter;
+mod frontend;
 mod multi_text;
 mod rlfmi;
-mod seal;
-mod search;
 mod suffix_array;
 mod util;
+mod wrapper;
 
-pub use crate::fm_index::FMIndex;
-pub use crate::rlfmi::RLFMIndex;
-
-pub use builder::SearchIndexBuilder;
+pub use backend::HeapSize;
```

Comment:
I feel it's a bit confusing that the `HeapSize` trait belongs to the backend module and is exposed. This trait is not only implemented by backend structs but also frontend structs.

Should we consider moving out this `HeapSize` to a dedicated module like `heap_size`? I'm not so much sure because such `heap_size` module will be very small.

Reply:
Ah, that's a good point, I should move `HeapSize`. Alternatively we get rid of `HeapSize` in the public API altogether and just expose a `heap_size` method on the various indexes. I think we may also be able to get rid of the trait then.

Reply:
See #61 for the plan